src/pages/api/twitter-request-token.ts

```diff
@@ -0,0 +1,80 @@
+import { createHmac, randomBytes } from "crypto"
+import { NextApiHandler } from "next"
+
+const CONSUMER_KEY = process.env.TWITTER_API_KEY
+const CONSUMER_SECRET = process.env.TWITTER_API_SECRET
+const SIGNATURE_METHOD = "HMAC-SHA1"
+const OAUTH_VERSION = "1.0"
+const DEFAULT_CALLBACK_URL = "https://guild.xyz/oauth"
+const BASE_URL = "https://api.twitter.com/oauth/request_token"
+const KEY = `${encodeURIComponent(CONSUMER_SECRET)}&`
+const RESPONSE_REGEX =
+  /oauth_token=(.*?)&oauth_token_secret=(.*?)&oauth_callback_confirmed=true/
```

Comment:
Is this a serverless endpoint? In that case this might be [ReDoS vulnerable](https://devina.io/redos-checker) too.

Reply:
I've made a similar change here, as I described in the core repo 👍 
https://github.com/agoraxyz/guild.xyz/pull/852/commits/fc3e224528edfa4f07ee222b19670b8476c63e80